common/src/main/java/com/wynntils/core/config/objects/ConfigHolder.java

```diff
@@ -16,7 +16,7 @@ public ConfigHolder(Object parent, Field field, Config metadata) {
                 parent,
                 field,
                 field.getType(),
-                field.getAnnotation(FeatureInfo.class).category(),
+                parent.getClass().getAnnotation(FeatureInfo.class).category(),
```

Comment:
This line assumes every feature has a @FeatureInfo, which is not enforced atm, so it crashes whenever it hits a feature like that. Fix that by defaulting the category to null if `parent.getClass().getAnnotation(FeatureInfo.class)` is `null`.

Reply:
Regarding this, do you think we should not allow loading features without a FeatureInfo annotation? This is not subject to this PR, but I think it might be helpful to do it in the long run.

Reply:
at the very least, we shouldn't load config options from features with no FeatureInfo annotation - it might be better to change the constructor to take a string for the category instead of determining it within the constructor. i plan on making a lot of changes/improvements wrt config loading soon, so this doesn't need to be dealt with in this PR

Reply:
It's probably not a secret that I don't think the FeatureInfo provides much value (at least not now -- I am not against introducing annotations for features if that ever changes in the future), and I am questioning their need. So I definitely do not want to see them enforced. :-)